org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacMethodBinding.java

```diff
@@ -117,8 +117,20 @@ public IJavaElement getJavaElement() {
 
 	@Override
 	public String getKey() {
-		// TODO Auto-generated method stub
-		throw new UnsupportedOperationException("Unimplemented method 'getKey'");
+		StringBuilder builder = new StringBuilder();
+		builder.append(this.getDeclaringClass().getName());
+		builder.append('.');
+		// TODO: what is a selector? why is it added?
+		for (var typeParam : this.methodSymbol.getTypeParameters()) {
+			builder.append(typeParam.getQualifiedName());
+		}
+		for (var param : this.methodSymbol.getParameters()) {
+			builder.append(param.getQualifiedName());
+		}
+		for (var thrownException : this.methodSymbol.getThrownTypes()) {
+			builder.append(thrownException.tsym.getQualifiedName());
+		}
+		return builder.toString();
```

Comment:
I'm kinda surprised the different parameters are not separated somehow. I'm curious to see if the ecj-based binding generates the same key.